src/app/main/component/ubs/components/ubs-order-details/ubs-order-certificate/ubs-order-certificate.component.ts

```diff
@@ -0,0 +1,367 @@
+  certDate: string;
+  certStatus: string;
+  failedCert = false;
+  object: {};
+  private destroy: Subject<boolean> = new Subject<boolean>();
+  public certificateError = false;
+  bonusesRemaining: boolean;
+  public locations: Locations[];
+  clickOnYes = true;
+  clickOnNo = true;
+
+  constructor(
+    private fb: FormBuilder,
+    private orderService: OrderService,
+    private shareFormService: UBSOrderFormService,
+    private localStorageService: LocalStorageService
+  ) {
+    this.initForm();
+  }
+
+  ngOnInit(): void {
+    localStorage.removeItem('UBSorderData');
+    this.orderService.locationSubject.pipe(takeUntil(this.destroy)).subscribe(() => {
+      if (this.localStorageService.getUbsOrderData()) {
+        this.calculateTotal();
+      }
+    });
+  }
+
+  disableAddCertificate() {
+    return this.certificates.length === this.formArrayCertificates.length;
+  }
+
+  initForm() {
+    this.orderDetailsForm = this.fb.group({
+      bonus: new FormControl('no'),
+      formArrayCertificates: this.fb.array([new FormControl('', [Validators.minLength(8), Validators.pattern(this.certificatePattern)])])
+    });
+  }
+
+  get formArrayCertificates() {
+    return this.orderDetailsForm.get('formArrayCertificates') as FormArray;
+  }
+
+  private certificateDateTreat(date: string) {
+    return date.split('-').reverse().join('-');
+  }
+
+  setNewValue(value: object) {
+    this.newItemEvent.emit(value);
+  }
+
+  sendDataToParents() {
+    const certificateObj = {
+      showCertificateUsed: this.showCertificateUsed,
+      certificateSum: this.fullCertificate,
+      displayCert: this.displayCert,
+      finalSum: this.finalSum,
+      pointsUsed: this.pointsUsed,
+      points: this.points,
+      isBonus: this.orderDetailsForm.get('bonus').value
+    };
+    this.setNewValue(certificateObj);
+  }
+
+  certificateMatch(cert): void {
+    if (cert.certificateStatus === CertificateStatus.ACTIVE || cert.certificateStatus === CertificateStatus.NEW) {
+      this.certificateSum += cert.certificatePoints;
+      this.displayCert = true;
+      this.addCert = true;
+    }
+    this.failedCert = cert.certificateStatus === CertificateStatus.EXPIRED || cert.certificateStatus === CertificateStatus.USED;
+    this.certificateSum = this.failedCert && this.formArrayCertificates.length === 1 ? 0 : this.certificateSum;
+    this.certDate = this.certificateDateTreat(cert.certificateDate);
+    this.certStatus = cert.certificateStatus;
+    this.fullCertificate = this.certificateSum;
+  }
+
+  private calculateTotal(): void {
+    this.total = 0;
+    this.bags.forEach((bag) => {
+      this.total += bag.price * bag.quantity;
+    });
+    this.showTotal = this.total;
+    if (this.total < this.minOrderValue && this.orderDetailsForm.dirty) {
+      this.displayMinOrderMes = true;
+      this.onSubmit = true;
+    } else {
+      this.displayMinOrderMes = false;
+      this.onSubmit = false;
+    }
+    this.finalSum = this.total - this.pointsUsed;
```

Comment:
please, replace if-else:
this.displayMinOrderMes = this.total < this.minOrderValue && this.orderDetailsForm.dirty;
this.onSubmit = this.displayMinOrderMes

Reply:
done